WordPress/gutenberg
# Fix Typography panel rendering from style hooks

This update fixes the Typography sidebar panel rendering, specifically if line-height/font-size style properties are disabled.

The Typography panel will no longer render if it does not contain any style controls.

## How has this been tested?

This has been tested locally within Gutenberg.

* Run `npm run dev`
* Add `add_theme_support( 'disable-custom-line-height' )` to your theme's `function.php`
* Add a Paragraph block
* Typography panel should still render, but without line height controls

* Remove font-sizes for Paragraph and/or Heading block
* Refresh
* Ensure that there is no Typography panel

## Screenshots

<img width="1090" alt="Screen Shot 2020-05-25 at 12 35 47 PM" src="https://user-images.githubusercontent.com/2322354/82831417-61b0bb80-9e86-11ea-99c1-c5811d7c44af.png">


## Types of changes

* Create a Typography panel component under `hooks/` that handles the rendering of typography style controls

## Checklist:

Resolves: https://github.com/WordPress/gutenberg/issues/22602
